.docs/05-headers-and-baggage.md

```diff
@@ -0,0 +1,11 @@
+[← Testing][back] - [:book:][index] - [Middleware →][next]
+
+# Headers & Baggage
```

Comment:
YARPC exposes RPC metadata, headers, and multi-hop headers (baggage). These are accessible and mutable in middleware through the OpenTracing context and transport.Request abstraction. Each encoding provides an API for getting metadata in handlers and setting metadata for callers.

Sections:

- Headers
- Baggage
- Tracing
- Caller Name
- Service Name
- Procedure Name
- Shard Key
- Routing Key
- Routing Delegate